astropy/units/quantity.py

```diff
@@ -685,7 +683,7 @@ def isscalar(self):
             (e.g. ``np.array(1)``), while this is True for quantities,
             since quantities cannot represent true numpy scalars.
         """
-        return not isiterable(self.value)
+        return not self.shape
```

Comment:
Ah.  I see how consistency with Numpy makes sense -- I just forgot it also worked that way.  Maybe add a sentence to the changelog: "This behavior is consistent with Numpy."